Installation issue: lfortran

### Steps to reproduce the issue

```console
$ spack spec -I lfortran
 -   lfortran@0.54.0~ipo+kokkos+llvm+stacktrace build_system=cmake build_type=Release generator=make platform=linux os=ubuntu24.04 target=zen2 %c,cxx,fortran=gcc@13.3.0
[e]      ^binutils@2.42~debuginfod~gas+gold~gprofng~headers~interwork~ld~libiberty~lto~nls~pgo+plugins build_system=autotools compress_debug_sections=zlib libs:=shared,static platform=linux os=ubuntu24.04 target=x86_64 
[e]      ^cmake@3.28.3~doc+ncurses+ownlibs~qtgui build_system=generic build_type=Release platform=linux os=ubuntu24.04 target=x86_64 
[+]      ^compiler-wrapper@1.0 build_system=generic platform=linux os=ubuntu24.04 target=zen2 
[e]      ^gcc@13.3.0+binutils+bootstrap~graphite+libsanitizer~mold~nvptx~piclibs~profiled~strip build_system=autotools build_type=RelWithDebInfo languages:='c,c++,fortran' platform=linux os=ubuntu24.04 target=x86_64 
[+]      ^gcc-runtime@13.3.0 build_system=generic platform=linux os=ubuntu24.04 target=zen2 
[e]      ^glibc@2.39 build_system=autotools platform=linux os=ubuntu24.04 target=x86_64 
[e]      ^gmake@4.3~guile build_system=generic platform=linux os=ubuntu24.04 target=x86_64 
[+]      ^kokkos@5.0.2~aggressive_vectorization~atomics_bypass~cmake_lang+complex_align~cuda~debug~debug_bounds_check+deprecated_code~hip_relocatable_device_code~hpx~hwloc~ipo~openmp~openmptarget~pic~rocm+serial+shared~sycl~tests~threads~tuning~wrapper build_system=cmake build_type=Release cxxstd=20 generator=make intel_gpu_arch=none platform=linux os=ubuntu24.04 target=zen2 %cxx=gcc@13.3.0
[+]      ^llvm@22.1.1+clang~cuda~flang~gold~ipo+libomptarget~libomptarget_debug~link_llvm_dylib+lld+lldb+llvm_dylib+lua~mlir+offload+polly~python~split_dwarf~utils~z3~zstd build_system=cmake build_type=Release compiler-rt=runtime generator=ninja libcxx=runtime libunwind=runtime openmp=runtime shlib_symbol_version=none targets:=aarch64,amdgpu,nvptx,x86 version_suffix=none platform=linux os=ubuntu24.04 target=zen2 %c,cxx=gcc@13.3.0
[+]          ^hwloc@2.13.0~cairo~cuda~gl~level_zero~libudev+libxml2~nvml~opencl+pci~rocm build_system=autotools libs:=shared,static patches:=b4db98b platform=linux os=ubuntu24.04 target=zen2 %c,cxx=gcc@13.3.0
[+]              ^libpciaccess@0.17 build_system=autotools platform=linux os=ubuntu24.04 target=zen2 %c=gcc@13.3.0
[+]                  ^util-macros@1.20.2 build_system=autotools platform=linux os=ubuntu24.04 target=zen2 
[+]          ^libedit@3.1-20240808 build_system=autotools platform=linux os=ubuntu24.04 target=zen2 %c=gcc@13.3.0
[+]          ^libffi@3.5.2 build_system=autotools platform=linux os=ubuntu24.04 target=zen2 %c,cxx=gcc@13.3.0
[+]          ^libxml2@2.15.1+pic~python+shared build_system=autotools platform=linux os=ubuntu24.04 target=zen2 %c=gcc@13.3.0
[+]              ^libiconv@1.18 build_system=autotools libs:=shared,static platform=linux os=ubuntu24.04 target=zen2 %c=gcc@13.3.0
[+]          ^lua@5.3.6+shared build_system=makefile fetcher=curl platform=linux os=ubuntu24.04 target=zen2 %c=gcc@13.3.0
[e]              ^curl@8.5.0+gssapi+ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs:=shared,static tls:=openssl platform=linux os=ubuntu24.04 target=x86_64 
[+]              ^readline@8.3 build_system=autotools patches:=21f0a03 platform=linux os=ubuntu24.04 target=zen2 %c=gcc@13.3.0
[+]              ^unzip@6.0 build_system=makefile patches:=179330d,24582ff,251d575,3371314,44599c8,47e9def,4e5a081,59c0983,64f6498,74bc961,7d8e5c7,81ca46c,881d2ed,aced0f2,b6f64d7,b7a14c3,c9a863e,ee9e260,f6f6236,f88b9d4,fde8f9d platform=linux os=ubuntu24.04 target=zen2 %c=gcc@13.3.0
[+]          ^ncurses@6.6~symlinks+termlib abi=none build_system=autotools patches:=7a351bc platform=linux os=ubuntu24.04 target=zen2 %c,cxx=gcc@13.3.0
[+]          ^ninja@1.13.2+re2c build_system=generic platform=linux os=ubuntu24.04 target=zen2 %c,cxx=gcc@13.3.0
[+]              ^re2c@4.4 build_system=autotools platform=linux os=ubuntu24.04 target=zen2 %c,cxx=gcc@13.3.0
[+]          ^perl-data-dumper@2.173 build_system=perl platform=linux os=ubuntu24.04 target=zen2 
[e]              ^perl@5.38.2~cpanm+opcode+open+shared+threads build_system=generic platform=linux os=ubuntu24.04 target=x86_64 
[e]          ^pkgconf@1.8.1 build_system=autotools platform=linux os=ubuntu24.04 target=x86_64 
[e]          ^python@3.12.3+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~optimizations+pic+pyexpat~pythoncmd+readline+shared+sqlite3+ssl~static~tests~tkinter+uuid+zlib build_system=generic platform=linux os=ubuntu24.04 target=x86_64 
[e]          ^swig@4.2.0 build_system=autotools platform=linux os=ubuntu24.04 target=x86_64 
[+]          ^xz@5.8.2~pic build_system=autotools libs:=shared,static platform=linux os=ubuntu24.04 target=zen2 %c=gcc@13.3.0
[+]      ^zlib-ng@2.3.3+compat+new_strategies+opt+pic+shared build_system=autotools platform=linux os=ubuntu24.04 target=zen2 %c,cxx=gcc@13.3.0

```


### Error message

<details><summary>Error message</summary>

<pre>
spack install lfortran %gcc
[+] /usr (external gcc-13.3.0-fhypd6ul4midmtovokp7kxc7bel6smrf)
[+] /usr (external glibc-2.39-thy5vqoaouxdwkh5hofjegtbqsl7r25d)
[+] /usr (external binutils-2.42-jbw5hw7ik6au2pd3iz6aqhck26aci5lj)
[+] /usr (external curl-8.5.0-qgrsi2iushneki7wyn7qiwuq4j6kv5j2)
[+] /usr (external swig-4.2.0-7sqh66lhlk4zfuvowbkgoq4ka7odd7wy)
[+] /usr (external cmake-3.28.3-bgezolbfmbil6ocsz56hhogkvh6a7klq)
[+] /opt/sophie/spack/linux-zen2/compiler-wrapper-1.0-6fgo5j4utkdwgdkjgz6yojfu4xggcck5
[+] /usr (external gmake-4.3-s3bjorvhflk6v6pwz6hfuaxdewixeyrb)
[+] /opt/sophie/spack/linux-zen2/gcc-runtime-13.3.0-kvi2mo6ifpygvapyv5kf7di3j27qtcwk
[+] /opt/sophie/spack/linux-zen2/zlib-ng-2.3.3-wetq3lnottzkddzv2wzdl4xfxxkjkpk5
[+] /opt/sophie/spack/linux-zen2/ncurses-6.6-zivoysas7esvlp3avaha2jdimu2ptd5r
[+] /opt/sophie/spack/linux-zen2/xz-5.8.2-qwjzeyxtdrr35kmocapvoyrre4zrc7fd
[+] /opt/sophie/spack/linux-zen2/libiconv-1.18-33msw5xfkoseaw7gqiziybpkjgpbjjnb
[+] /opt/sophie/spack/linux-zen2/libpciaccess-0.17-gfpodsihfmtgsndnwzry6o7alnwgjmxs
[+] /opt/sophie/spack/linux-zen2/unzip-6.0-32hrq6b3irluoj63fncowam7fnufrx6g
[+] /opt/sophie/spack/linux-zen2/kokkos-5.0.2-fhxcyroakouvhm5hfbklylnhnxqheyvf
[+] /opt/sophie/spack/linux-zen2/libffi-3.5.2-uwrte3yrpr5levbfrfkt5kknhaoyr672
[+] /opt/sophie/spack/linux-zen2/readline-8.3-np4bzcwqzes3pialnashieoeex4vvbgv
[+] /opt/sophie/spack/linux-zen2/libedit-3.1-20240808-etm5sobtmdgnf7unsn26dary4ue2j74n
[+] /opt/sophie/spack/linux-zen2/libxml2-2.15.1-as5xy7obie7j6liohzjqtgk55jy5lkpv
[+] /opt/sophie/spack/linux-zen2/lua-5.3.6-ksquxttta6qpt6z247gevpgojazquxeh
[+] /opt/sophie/spack/linux-zen2/hwloc-2.13.0-657dn24zc373tcgzxzy6mdtscyspia64
[+] /opt/sophie/spack/linux-zen2/llvm-22.1.1-bx4mlmpd7fklios7gme6fj7kkonwagfl
==> No binary for lfortran-0.54.0-k5aersqslkvlfvge3ocjlf4c65fhbzwx found: installing from source
==> Installing lfortran-0.54.0-k5aersqslkvlfvge3ocjlf4c65fhbzwx [24/24]
==> Using cached archive: /root/spack/var/spack/cache/_source-cache/archive/a4/a46c44f8398ed0d14ca051a08982a3001642449c06a3be1c30944c3e027bbf51.tar.gz
==> No patches needed for lfortran
==> lfortran: Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    '/usr/bin/cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/opt/sophie/spack/linux-zen2/lfortran-0.54.0-k5aersqslkvlfvge3ocjlf4c65fhbzwx' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=ON' '-DCMAKE_INSTALL_RPATH:STRING=/opt/sophie/spack/linux-zen2/lfortran-0.54.0-k5aersqslkvlfvge3ocjlf4c65fhbzwx/lib;/opt/sophie/spack/linux-zen2/lfortran-0.54.0-k5aersqslkvlfvge3ocjlf4c65fhbzwx/lib64' '-DCMAKE_PREFIX_PATH:STRING=/opt/sophie/spack/linux-zen2/compiler-wrapper-1.0-6fgo5j4utkdwgdkjgz6yojfu4xggcck5;/opt/sophie/spack/linux-zen2/kokkos-5.0.2-fhxcyroakouvhm5hfbklylnhnxqheyvf;/opt/sophie/spack/linux-zen2/llvm-22.1.1-bx4mlmpd7fklios7gme6fj7kkonwagfl;/opt/sophie/spack/linux-zen2/zlib-ng-2.3.3-wetq3lnottzkddzv2wzdl4xfxxkjkpk5;/opt/sophie/spack/linux-zen2/gcc-runtime-13.3.0-kvi2mo6ifpygvapyv5kf7di3j27qtcwk' '-DCMAKE_BUILD_TYPE:STRING=Release' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_POLICY_DEFAULT_CMP0090:STRING=NEW' '-DCMAKE_FIND_USE_PACKAGE_REGISTRY:BOOL=OFF' '-DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=ON' '-DWITH_LLVM:BOOL=ON' '-DWITH_STACKTRACE:BOOL=ON' '-DWITH_KOKKOS:BOOL=ON' '-DLFORTRAN_BUILD_ALL=yes' '/tmp/root/spack-stage/spack-stage-lfortran-0.54.0-k5aersqslkvlfvge3ocjlf4c65fhbzwx/spack-src'

2 errors found in build log:
  >> 3    /usr/bin/bash: build0.sh: No such file or directory
  >> 4    CMake Error at CMakeLists.txt:12 (message):
     5      Running build0.sh failed (see error above)
     6    
     7    
     8    -- Configuring incomplete, errors occurred!

See build log for details:
  /tmp/root/spack-stage/spack-stage-lfortran-0.54.0-k5aersqslkvlfvge3ocjlf4c65fhbzwx/spack-build-out.txt


</pre></details>


### Information on your system

# spack debug report
* **Spack:** 1.2.0.dev0 (https://github.com/spack/spack/commit/e0997bb279d5cc1e1a243dcd91e55ac63089ff29)
* **Builtin repo:** https://github.com/spack/spack-packages/commit/6996192cd696564eaaa4e9c87ab39d3b899bf617
* **Python:** 3.12.3
* **Platform:** linux-ubuntu24.04-zen2
# spack maintainers lfortran
@certik

### Additional information

[spack-build-env.txt](https://github.com/user-attachments/files/26093591/spack-build-env.txt)
[spack-build-out.txt](https://github.com/user-attachments/files/26093592/spack-build-out.txt)

### General information

- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

--- a/repos/spack_repo/builtin/packages/lfortran/package.py
+++ b/repos/spack_repo/builtin/packages/lfortran/package.py
@@ -19,6 +19,7 @@ class Lfortran(CMakePackage):
 
     # The build process uses 'git describe --tags' to get the package version
     version("main", branch="main", get_full_repo=True)
+    version("0.61.0", sha256="e832c1d76c371da7a7e11ef9e7b686d9047788136dcfb20093da5dc165fcd20f")
     version("0.54.0", sha256="a46c44f8398ed0d14ca051a08982a3001642449c06a3be1c30944c3e027bbf51")
     version("0.49.0", sha256="a9225fd33d34ce786f72a964a1179579caff62dd176a6a1477d2594fecdc7cd6")
     version("0.30.0", sha256="aafdfbfe81d69ceb3650ae1cf9bcd8a1f1532d895bf88f3071fe9610859bcd6f")
@@ -38,6 +39,7 @@ class Lfortran(CMakePackage):
 
     depends_on("python@3:", type="build", when="@main")
     depends_on("cmake", type="build")
+    depends_on("cmake@3.22:", type="build", when="+kokkos")
     depends_on("kokkos", type=("build", "run"), when="+kokkos")
     depends_on("llvm@11:15", type=("build", "run"), when="@0.19.0+llvm")
     depends_on("llvm@11:16", type=("build", "run"), when="@0.30.0+llvm")
@@ -46,6 +48,8 @@ class Lfortran(CMakePackage):
     depends_on("re2c", type="build", when="@main")
     depends_on("bison@:3.4", type="build", when="@main")
     depends_on("binutils@2.38:", type="build", when="platform=linux")
+    depends_on("zstd")
+    depends_on("libunwind")
 
     def cmake_args(self):
         args = [
@@ -54,7 +58,7 @@ class Lfortran(CMakePackage):
             self.define_from_variant("WITH_KOKKOS", "kokkos"),
         ]
 
-        if self.spec.satisfies("@0.54.0:"):
+        if self.spec.satisfies("@main"):
             args.append("-DLFORTRAN_BUILD_ALL=yes")
 
         return args

--- a/repos/spack_repo/builtin/packages/lfortran/package.py
+++ b/repos/spack_repo/builtin/packages/lfortran/package.py
@@ -58,6 +58,7 @@ class Lfortran(CMakePackage):
             self.define_from_variant("WITH_KOKKOS", "kokkos"),
         ]
 
+        # Only call bootstrap script for git checkout
         if self.spec.satisfies("@main"):
             args.append("-DLFORTRAN_BUILD_ALL=yes")
 

